src/Checkbox/CheckboxShared.js

```diff
@@ -1,28 +1,98 @@
 // @flow
 
 import * as React from 'react';
+import { View, TouchableWithoutFeedback } from 'react-native';
 import CheckboxText from './CheckboxText';
 import CheckboxIcon from './CheckboxIcon';
+import Hoverable from './Hoverable';
+import theme, { parseStringToFloat } from './styles';
+import StyleSheet from '../PlatformStyleSheet';
+import type { CheckboxSharedProps, CheckboxSharedState } from './CheckboxTypes';
 
-type Props = {|
-  +label?: React.Node,
-  +hasError?: boolean,
-  +disabled?: boolean,
-  +checked?: boolean,
-  +info?: React.Node,
-|};
-
-export default function CheckboxShared({
-  label,
-  hasError,
-  disabled,
-  checked,
-  info,
-}: Props) {
-  return (
-    <>
-      <CheckboxIcon checked={checked} hasError={hasError} disabled={disabled} />
-      <CheckboxText label={label} checked={checked} info={info} />
-    </>
-  );
+class CheckboxShared extends React.Component<
+  CheckboxSharedProps,
+  CheckboxSharedState
+> {
+  constructor(props: CheckboxSharedProps) {
+    super(props);
+
+    this.state = {
+      showFocus: false,
```

Comment:
```suggestion
      focused: false,
```
maybe to be a bit more consistent

Reply:
I've used `focused` name at first. But I've changed it to prevent confusion with actual focus state of component. Since it serves only for switching visual style of checkbox focus border.
Which is turned off by clicking on component, event that the component is still in focus (`focused`).
Hope it makes sense, basically it just prevents checkbox being stuck with thick blue border when focused-on with keyboard and then clicked.

Reply:
I see. So maybe the name should be changed to `focusDisplayed` or something similar. What do you think?

Reply:
Definitely, `focusDisplayed` sounds good. Lot better than original function-like `showFocus`.

Updated